src/index.js

```diff
@@ -270,10 +270,35 @@ function statelessComponentHook(Ctor) {
 	return Wrapped;
 }
 
+function validatePropTypes (vnode) {
+	if (DEV) {
+		let componentClass = typeof vnode.nodeName === "function"
+			? vnode.nodeName
+			: vnode.type;
+
+		if (typeof componentClass !== "function") return;
+		let name = componentClass.displayName || componentClass.name;
+		let propTypes = componentClass.propTypes;
+		if (propTypes) {
+			let props = vnode.props;
+			if (
+				!(vnode.props && vnode.props.children) &&
+				vnode.children &&
+				vnode.children.length
+			) {
+				props.children = vnode.children;
+			}
+			propsHook(props);
+			PropTypes.checkPropTypes(propTypes, props, 'prop', name);
+		}
+	}
+}
 
 function createElement(...args) {
 	upgradeToVNodes(args, 2);
-	return normalizeVNode(h(...args));
+	let node = h(...args);
+	validatePropTypes(node);
```

Comment:
Just had a thought (I think I was noodling on this when I failed to review before):

What if we moved the invocation of this validation into [`handleComponentVNode()`](https://github.com/developit/preact-compat/blob/master/src/index.js#L106-L113)?  That would make proptype validation work in non-compat components if you install compat, which is kinda compelling.  Also it would nicely reuse the existing logic around detecting and forking for component VNodes vs element VNodes.

Reply:
I honestly got this working in a very specific way and I don't remember all the details. I do remember that I had good reasons for doing it the way I did though 😅